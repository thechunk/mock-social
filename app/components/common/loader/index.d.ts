interface ILoader {
    loading: boolean,
    style: import('react-native').StyleProp<import('react-native').ViewStyle>
}
