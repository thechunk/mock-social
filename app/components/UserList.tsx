import React, {PureComponent} from 'react';
import UserListCell from "./UserListCell";
import {IWithUserListDataProps} from "../hoc/withUserListData";

class UserList extends PureComponent<IWithUserListDataProps> {
    render() {
        return this.props.users != null
            ? this.props.users.map(v => <UserListCell key={v.id} title={v.name} />)
            : null;
    }
}

export default UserList;