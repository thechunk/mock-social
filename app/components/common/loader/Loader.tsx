import * as React from 'react';
import {PureComponent} from "react";
import {ActivityIndicator, ActivityIndicatorProps, View} from "react-native";
import styles from './styles';

export default class Loader extends PureComponent<ILoader & ActivityIndicatorProps> {
    render() {
        return this.props.loading ? (
            <View style={[styles.loader, this.props.style]}>
                <ActivityIndicator size="small" {...this.props as ActivityIndicatorProps} />
            </View>
        ) : null;
    }
}