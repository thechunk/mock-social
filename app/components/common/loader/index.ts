/// <reference path='./index.d.ts' />

export {default as Loader} from './Loader';

