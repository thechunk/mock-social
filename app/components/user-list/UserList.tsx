import React, {PureComponent} from 'react';
import {TitleSubtitleListCell} from '../common/list';
import withUserListData from '../../hoc/withUserListData';
import {FlatList, GestureResponderEvent} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {NavigationScreenProps, withNavigation} from 'react-navigation';
import {inherited} from './styles';
import {Loader} from "../common/loader/index";
import g, {Insets, Color} from "../../styles/global";

class UserList extends PureComponent<IWithUserListDataProps & NavigationScreenProps> {
    navigateToDetail(user: IUser) {
        return (event: GestureResponderEvent) => {
            this.props.navigation.navigate('UserDetail',
                { hydrate: user, id: user.id } as IWithUserDataOptionalProps)
        }
    }

    render() {
        return this.props.loading ? <Loader loading={this.props.loading} style={inherited.loader} /> : (
            <FlatList
                contentContainerStyle={g.scrollViewHeaderOffset}
                scrollIndicatorInsets={Insets.ScrollViewWithFloatingHeader}
                data={this.props.users}
                keyExtractor={v => v.id.toString()}
                renderItem={v => (
                    <TitleSubtitleListCell
                        key={v.index}
                        title={v.item.username}
                        subtitle={v.item.name}
                        rightAccessory={() => (
                            <Icon name="ios-arrow-dropright" size={30} color={Color.SecondaryAccent} />
                        )}
                        onPress={this.navigateToDetail(v.item)}
                        cellStyle={[inherited.cell]}
                        titleStyle={[inherited.title]}
                        subtitleStyle={[inherited.subtitle]} />
                )} />
        );
    }
}

export default withNavigation(withUserListData(UserList));