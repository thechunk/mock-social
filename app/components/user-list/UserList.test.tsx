import * as React from 'react';
import UserList from './UserList';

import {shallow} from 'enzyme';
import {Loader} from "../common/loader/index";

describe('UserList', () => {
    it('renders correctly', () => {
        expect(shallow(<UserList users={[]} />)).toMatchSnapshot();
    });

    it('navigateToDetail navigates away with data', () => {
    });

    it('shows loader', () => {
        const component = shallow(<UserList users={[]} navigation={{}} loading={true} />);
    });

    it('shows content', () => {
        const component = shallow(<UserList users={[]} navigation={{}} loading={false} />);
    })
});
