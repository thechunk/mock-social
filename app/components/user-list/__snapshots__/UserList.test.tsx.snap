// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList renders correctly 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
